source/microbit/microbitmusic.cpp

```diff
@@ -43,97 +43,94 @@ extern "C" {
 #define DEFAULT_DURATION 4 // Crotchet
 #define ARTICULATION_MS  10 // articulation between notes in milliseconds
 
-typedef struct _music_state_t {
+typedef struct _music_data_t {
     uint16_t bpm;
     uint16_t ticks;
 
     // store these to simplify the writing process
     uint8_t last_octave;
     uint8_t last_duration;
-} music_state_t;
+
+    // Asynchronous parts.
+    volatile uint8_t async_state;
+    uint32_t async_wait_ticks;
+    bool async_loop;
```

Comment:
This entire struct can be made smaller (by 4 bytes) by moving this async_loop member up one, to sit just after async_state.